proto/internal/temporal/server/api/persistence/v1/task_queues.proto

```diff
@@ -111,14 +113,42 @@ message DeploymentData {
     // Set of deployments that this task queue belongs to.
     // Current deployment is defined implicitly as the deployment with the most recent
     // TaskQueueData.last_became_current_time.
-    repeated DeploymentDataItem deployments = 1;
-
+    // Deprecated.
+    repeated DeploymentDataItem deployments = 1 [deprecated = true];
+
+    // Set of deployment versions that this task queue belongs to.
+    // Current Version is defined implicitly as the version with `is_current==true` and the most
+    // recent `routing_update_time`.
+    // Ramping Version is defined implicitly as the version with `ramp_percentage>0` and the most
+    // recent `routing_update_time`.
+    // The Ramping Version receives a share of unversioned/unpinned tasks according to its
+    // `ramp_percentage`. If there is no Ramping Version, all the unversioned/unpinned tasks are
+    // routed to the Current Version. If there is no Current Version, any poller with UNVERSIONED
+    // (or unspecified) WorkflowVersioningMode will receive the tasks.
+    repeated DeploymentVersionData versions = 2;
+
+    // Deprecated.
     message DeploymentDataItem {
         temporal.api.deployment.v1.Deployment deployment = 1;
         temporal.server.api.deployment.v1.TaskQueueData data = 2;
     }
 }
 
+// The source of truth for this data is in the WorkerDeployment entity workflows, which is
+// synced to all TQs whenever the source changes.
+message DeploymentVersionData {
+    // Special case: `version.version` may be absent (but not `version.deployment_name`). That is
+    // only valid when `ramp_percentage` is non-zero, representing ramping to unversioned in a
+    // particular deployment name.
+    temporal.api.deployment.v1.WorkerDeploymentVersion version = 1;
+    // Last time `is_current` or `ramp_percentage` of this version changed.
+    google.protobuf.Timestamp routing_update_time = 2;
+    // If this version is the current version of its deployment.
+    bool is_current = 3;
+    // Range: [0, 100]. Must be zero if is_current is true. Must be non-zero if version is null.
```

Comment:
I don't understand what you mean by `Must be non-zero if version is null.`. It seems like `DeploymentVersionData.version == nil` is invalid. If we are ramping to unversioned, we still need the deployment name inside `version`.